test/test_utils.ts

```diff
@@ -1,21 +1,24 @@
 import sinon from "sinon";
 import { expect } from "chai";
 
+export function fakeHeaders(headers: { [key: string]: string | number } = {}) {
+  const h = new (global as any).Headers();
+  Object.entries(headers).forEach(([k, v]) => h.set(k, v));
+  return h;
+}
+
 export function fakeServerResponse(
   status: number,
   json: any,
   headers: { [key: string]: string | number } = {}
 ) {
+  const respHeaders = fakeHeaders(headers);
+  if (!respHeaders.has("Content-Length")) {
+    respHeaders.set("Content-Length", JSON.stringify(json).length);
+  }
   return Promise.resolve({
```

Comment:
Since TypeScript already has a global declaration for `Headers`, we don't need to pull it off of `global`. Also, by casting `global` to `any`, the resulting `Headers` is _also_ any, meaning the calls to `has` and `set` also lose typechecking.

I tried to propose this as a suggested change, but the diff was difficult to follow since the line numbers are affected by Prettier's formatting. I also didn't want to risk giving you something that ends up failing the formatting checks!

I've marked the changed lines with an asterisk:

```ts
export function fakeHeaders(headers: { [key: string]: string | number } = {}) {
  const h = new Headers(); // * don't cast global as any
  Object.entries(headers).forEach(([k, v]) =>
    h.set(k, typeof v === "number" ? v.toString() : v) // * convert numbers to strings
  );
  return h;
}

export function fakeServerResponse(
  status: number,
  json: any,
  headers: { [key: string]: string | number } = {}
) {
  const respHeaders = fakeHeaders(headers);
  if (!respHeaders.has("Content-Length")) {
    respHeaders.set("Content-Length", JSON.stringify(json).length.toString()); // * convert number to string
  }
  return Promise.resolve({
    status: status,
    headers: respHeaders,
    text() {
      return Promise.resolve(JSON.stringify(json));
    },
  });
}
```